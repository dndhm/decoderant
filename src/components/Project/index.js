import React from 'react';
import propTypes from 'prop-types';
import injectStaticSheet from 'inject-static-sheet';

import colors, {
	PROJECT_SLATE,
	WHITE,
} from '../../constants/colors';

import Heading from '../Heading';
import Paragraph from '../Paragraph';
import ProjectTags, { tags } from '../ProjectTags';

const styles = {
	container: {
		background: colors[PROJECT_SLATE],
		'text-align': 'center',
	},
	content: {
		padding: '10px',
	},
	description: {
		margin: '0 auto 10px',
		width: '80%',
	},
};

const Project = ({
	activeTag,
	classes,
	description,
	tags,
	title,
}) => (
	<div className={classes.container}>
		<div className={classes.content}>
			<Heading color={WHITE} level={3}>{title}</Heading>
			<ProjectTags active={activeTag} tags={tags} />
			<div className={classes.description}>
				<Paragraph>{description}</Paragraph>
			</div>
		</div>
	</div>
);

Project.propTypes = {
	activeTag: propTypes.oneOf(tags),
	classes: propTypes.object,
	description: propTypes.node.isRequired,
	tags: propTypes.arrayOf(
		propTypes.oneOf(tags)
	).isRequired,
	title: propTypes.string.isRequired,
};

export default injectStaticSheet(styles, 'Project')(Project);
