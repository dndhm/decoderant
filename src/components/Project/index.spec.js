import React from 'react';
import { create } from 'react-test-renderer';

import Project from './';

describe('Project', () => {
	test('should render correctly', () => {
		const tree = create(
			<Project
				description="This is my website."
				tags={['React', 'NodeJS']}
				title="DECODERANT"
			/>
		).toJSON();

		expect(tree).toMatchSnapshot();
	});
});
