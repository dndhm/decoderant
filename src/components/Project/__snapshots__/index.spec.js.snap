// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Project should render correctly 1`] = `
<div
  className="Project--container"
>
  <div
    className="Project--content"
  >
    <h3
      className="Heading--heading Heading--h3 Heading--color-white"
    >
      DECODERANT
    </h3>
    <ul
      className="ProjectTags--list"
    >
      <li
        className="ProjectTags--listItem"
      >
        React
      </li>
      <li
        className="ProjectTags--listItem"
      >
        NodeJS
      </li>
    </ul>
    <div
      className="Project--description"
    >
      <p
        className="Paragraph--paragraph"
      >
        This is my website.
      </p>
    </div>
  </div>
</div>
`;
